include/openssl/err.h

```diff
@@ -38,19 +38,9 @@ extern "C" {
 # define ERR_TXT_MALLOCED        0x01
 # define ERR_TXT_STRING          0x02
 
-# define ERR_FLAG_MARK           0x01
-# define ERR_FLAG_CLEAR          0x02
-
-# define ERR_NUM_ERRORS  16
-typedef struct err_state_st {
-    int err_flags[ERR_NUM_ERRORS];
-    unsigned long err_buffer[ERR_NUM_ERRORS];
-    char *err_data[ERR_NUM_ERRORS];
-    int err_data_flags[ERR_NUM_ERRORS];
-    const char *err_file[ERR_NUM_ERRORS];
-    int err_line[ERR_NUM_ERRORS];
-    int top, bottom;
-} ERR_STATE;
+# if !OPENSSL_API_3
+#  include <openssl/err_state.h>
```

Comment:
I don't understand why you make a new public file to deprecated something. Why don't you just put the `#if` around it, instead of moving it to a new file?

Reply:
Because `crypto/err/err.c` still needs the definition.  The easiest was to place it in another file that is unguarded.  Th other solution would be to duplicate the definition in a private header, but that would be a bad solution.

Reply:
Another solution is to have an internal macro defined by `crypto/err/err.c`, say `OPENSSL_FORCEDEF_ERR_STATE`, which overrides deprecation.  Is that preferable?

Reply:
I was typing that in my browser when your reply showed up.  :) FWIW, *I* think that's preferable.

Reply:
Okie.  Maybe...  Yesterday, I thought *that* was a bad idea, but I'm less sure today.

Reply:
Done